synapse/util/caches/response_cache.py

```diff
@@ -54,6 +58,20 @@ class ResponseCacheContext(Generic[KV]):
     """
 
 
+@attr.s(auto_attribs=True)
+class ResponseCacheEntry:
+    result: AbstractObservableDeferred
+    """The (possibly incomplete) result of the operation.
+
+    Note that we continue to store an ObservableDeferred even after the operation
+    completes (rather than switching to an immediate value), since that makes it
+    easier to cache Failure results.
+    """
+
+    opentracing_span_context: Any
```

Comment:
Is there a better type we can use for opentracing spans or is this awaiting some of the work that @H-Shay is doing?